zebra/connected.c

```diff
@@ -463,6 +497,16 @@ void connected_down(struct interface *ifp, struct connected *ifc)
 	rib_delete(afi, SAFI_MULTICAST, zvrf->vrf->vrf_id, ZEBRA_ROUTE_CONNECT,
 		   0, 0, &p, NULL, &nh, 0, zvrf->table_id, 0, 0, false);
 
+	if (remove_local) {
+		rib_delete(afi, SAFI_UNICAST, zvrf->vrf->vrf_id,
```

Comment:
What happens if you operationally try to change the prefix length without breaking connectivity?

e.g. start from
```
interface eth0
 ip address 192.168.1.1/24
```
⇒ user notices network is not large enough
⇒ tries to add new addr before deleting old:
```
ip address 192.168.1.1/23
no ip address 192.168.1.1/24
```

(this should also be a topotest?)

Reply:
If run from w/ in FRR it just replaces the /24 with the /23:
```
!
interface dummy1
 ip address 192.168.30.1/23
exit
!```

Reply:
tested w/ David looking over my shoulder we are happy with behavior now